System/Posix/Terminal/Common.hsc

```diff
@@ -352,40 +447,132 @@ withMinInput termios count = unsafePerformIO $ do
     let c_cc = (#ptr struct termios, c_cc) p
     pokeElemOff c_cc (#const VMIN) (fromIntegral count :: CCc)
 
-data BaudRate
-  -- These are the standard rates stipulated by POSIX:
-  = B0
-  | B50
-  | B75
-  | B110
-  | B134
-  | B150
-  | B200
-  | B300
-  | B600
-  | B1200
-  | B1800
-  | B2400
-  | B4800
-  | B9600
-  | B19200
-  | B38400
-  -- These are non-standard rates that are often present on modern Unixes:
-  | B57600
-  | B115200
-  | B230400
-  | B460800
-  | B500000
-  | B576000
-  | B921600
-  | B1000000
-  | B1152000
-  | B1500000
-  | B2000000
-  | B2500000
-  | B3000000
-  | B3500000
-  | B4000000
+newtype BaudRate = BaudRate CSpeed
+
+#ifdef B0
+pattern B0 :: BaudRate
+pattern B0 = BaudRate (#const B0)
+#endif
```

Comment:
Here and elsewhere I'd like to see the baud rates that are not supported on the current platform mapped to a known unused value, likely `-1` will work, ... so that the `well-known` values are defined on all platforms, even if not actually mapped to usable baud rate.

The sentinel value should also have an explicit name (`UnsupportedBaudRate` or similar), which allows users to check whether the baud-rate they are looking for is supported or not.

Reply:
A sentinel value works much better than bottom

Reply:
What do you think is defined to be bottom here? On a given platform, the undefined baud rate macros' patterns are also undefined.

Reply:
My point is that none of the values should be "error", they should instead be a non-error sentinel, that one can compare a given value with to determine whether it is supported or not.  The unsupported value can probably be (-1).

Reply:
With this PR, none of the values are "error."

Reply:
But the patterns should be defined unconditionally, even when the OS does not have a define for the name, in which case the name should map to a sentinel value that indicates an invalid Baud rate.

Reply:
Why is that better than the behavior one gets with C, where using an undefined baud macro would yield a compile time error instead?

Reply:
> Why is that better than the behavior one gets with C, where using an undefined baud macro would yield a compile time error instead?

Because, in an actual C application, the call site would have an `#Ifdef` the particular macro, and the code would compile on all platforms!  Here, we're providing a Haskell module that abstracts the C interface, but the caller is not expected to have feature-test `CPP` macros in their code, that's our job. And yet the caller's code must compile on all platforms.

Therefore, we must present the same set of pattern synonyms on all platforms, and allow run-time discovery of which ones are supported by the platform.